public/components/notebooks/components/main.tsx

```diff
@@ -194,27 +211,76 @@ export class Main extends React.Component<MainProps, MainState> {
   };
 
   // Deletes existing notebooks
-  deleteNotebook = (notebookList: string[], toastMessage?: string) => {
-    return this.props.http
-      .delete(`${NOTEBOOKS_API_PREFIX}/note/${notebookList.join(',')}`)
-      .then((res) => {
+  deleteNotebook = async (notebookList: string[], toastMessage?: string) => {
+    const deleteNotebook = (id: string) => {
+      const isValid = isValidUUID(id);
+      const route = isValid
+        ? `${NOTEBOOKS_API_PREFIX}/note/savedNotebook/${id}`
+        : `${NOTEBOOKS_API_PREFIX}/note/${id}`;
+      return this.props.http.delete(route).then((res) => {
         this.setState((prevState) => ({
-          data: prevState.data.filter((notebook) => !notebookList.includes(notebook.id)),
+          data: prevState.data.filter((notebook) => notebook.id !== id),
         }));
+        return res;
+      });
+    };
+
+    const promises = notebookList.map((id) =>
+      deleteNotebook(id).catch((err) => {
+        this.setToast(
+          'Error deleting notebook, please make sure you have the correct permission.',
+          'danger'
+        );
+        console.error(err.body.message);
+      })
+    );
+
+    Promise.all(promises)
+      .then(() => {
         const message =
           toastMessage || `Notebook${notebookList.length > 1 ? 's' : ''} successfully deleted!`;
         this.setToast(message);
-        return res;
+      })
+      .catch((err) => {
+        console.error('Error in deleting multiple notebooks', err);
```

Comment:
this catch seems redundent since each promise has its own catch. if a notebook failed to be deleted, the error will be caught, then the promises will still go to `.then` and add toast of successfully deleted?

you might want to use Promise.allSettled to process results at the end regardless of failures